tests/test_codelist.py

```diff
@@ -203,29 +203,38 @@ def test_to_yaml_from_directory(tmp_path):
     )
 
 
-def test_RegionCodeList_hierarchy_filter():
+def test_RegionCodeList_filter():

```

Comment:
Do you need the extra attributes here if you're just testing for the hierarchy filtering?
I don't think you're actually using them to test anything so I would cut them out here and of the region.yaml file.
`hierarchy` and `file` you can just add when you're creating the Code objects making for a shorter test.

Reply:
Disagree, the "file" attribute is added automatically at initialization of the **RegionCodeList** instance from a file - so you have to add it to ensure equality between expected and observed objects.